pkg/config/manager.go

```diff
@@ -101,6 +101,10 @@ func NewManager() *Manager {
 func (m *Manager) GetConfig(key string) (string, error) {
 	m.RLock()
 	defer m.RUnlock()
+	return m.getConfig(key)
```

Comment:
I would suggest to use a sync.map
rather than add locks on top of all the maps.

Reply:
Try to avoid using unnecessary locks is virtue.